src/main/java/com/google/sps/servlets/ListOrganizationsServlet.java

```diff
@@ -72,25 +74,47 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public Query getQueryFromParams(HttpServletRequest request, GivrUser currentUser) {
     Query query = new Query("Distributor").addSort("creationTimeStampMillis", SortDirection.DESCENDING);
 
+    ArrayList<Filter> filterCollection = new ArrayList<Filter>();
+
+    String zipcode  = "";
+    boolean queryForZipcode = false;
+    if (request.getParameter("zipcode") != null) {
+      zipcode = request.getParameter("zipcode");
+      queryForZipcode = true;
+    }
+
     /* displayUserOrgsParameter is true when user only wants to see orgs they moderate*/
     String displayUserOrgsParameter = request.getParameter("displayUserOrgs");
     boolean displayUserOrgs = coerceParameterToBoolean(request, displayUserOrgsParameter);
     boolean isUserLoggedIn = (currentUser != null);
     // Ternary operator is used to check if userIsMaintainer to protect against null currentUser
     boolean userIsMaintainer = isUserLoggedIn ? currentUser.isMaintainer() : false;
 
+    if (queryForZipcode) {
+      filterCollection.add(new FilterPredicate("orgZipCode", FilterOperator.EQUAL, zipcode));
+    }
+
     if (isUserLoggedIn && displayUserOrgs) {
       /* If the user is logged in and wants to just see their orgs, get their user ID & index with it*/
       String userId = currentUser.getUserId();
-      query.setFilter(new FilterPredicate("moderatorList", FilterOperator.EQUAL, userId));
+      filterCollection.add(new FilterPredicate("moderatorList", FilterOperator.EQUAL, userId));
     }
 
-    if (!userIsMaintainer) {
+    if (userIsMaintainer) {
```

Comment:
The comment below and this check don't agree - it states that you want non-maintainers to only see approved orgs.

Reply:
This is getting complicated enough that you should consider a test:

See:

https://cloud.google.com/appengine/docs/standard/java/tools/localunittesting#datastore-memcache

Reply:
You're right, this should be (!userIsMaintainer), I changed it when trying to debug something and forgot to change it back. I agree that unit testing would definitely be useful for this, and I'm going to implement it for this servlet